src/main.qml

```diff
@@ -17,27 +17,205 @@
 
 import QtQuick 2.9
 import QtSensors 5.11
+import QtGraphicalEffects 1.15
 import org.asteroid.controls 1.0
+import Nemo.KeepAlive 1.1
 
 Application {
     id: app
 
     centerColor: "#b04d1c"
     outerColor: "#421c0a"
 
-    property int bpm: 0;
+    readonly property real arcStartOffset: -181
+    readonly property real arcGapHeart: 52
+
+    property real arcBpmGap: app.lastBpm > 0 ? 28 : 0
+    property real arcBpmOffset: app.lastBpm > app.bpm ? 75 : 15
+
+    property real pulseAnimationDuration: app.bpm === 0 ?
+                         100 :
+                         1000 / (app.bpm / 20)
+    property real arcAnimationDuration: app.bpm === 0 ?
+                         30000 :
+                         1000 / (app.bpm / 20)
+
+
+    property real arcEnd: arcStart
+    property real arcStart: arcStartOffset + arcGapHeart/2 - pulseWidthArc
+    property int bpm: 0
+    property int lastBpm: 0
+    property real lastBpmAngle: app.lastBpm > app.bpm ? 30 : -30
+    property bool pulseToggle: true
+
+    property int pulseWidthArc: pulseToggle ?
+                                 0 :
+                                 -6
+    property int pulseWidth: pulseToggle ?
+                                 app.height * 0.20 :
+                                 app.height * 0.26
+
+    onArcEndChanged: canvas.requestPaint()
+    onBpmChanged: {
+        app.arcEnd = Qt.binding(function() { return arcStartOffset + (360 - arcGapHeart) + 28 + pulseWidthArc })
```

Comment:
This can also be moved to the onCompleted function if I'm not mistaken (Removing the other `app.arcEnd` line).